activerecord/lib/active_record/integration.rb

```diff
@@ -100,7 +100,7 @@ def to_param(method_name = nil)
           define_method :to_param do
             if (default = super()) &&
                  (result = send(method_name).to_s).present? &&
-                   (param = result.squish.truncate(20, separator: /\s/, omission: nil).parameterize).present?
+                   (param = result.squish.parameterize.truncate(20, separator: /-/, omission: '')).present?
```

Comment:
I think it would be safer to keep parameterize towards the end to not cause any side effects from truncate affect parameterization.